azavea/python-omgeo
# CI using GitHub Actions

## Overview

travis-ci.org shut down, and migrating to GitHub Actions seemed easy and is our preferred solution anyway, so this makes the switch.

## Notes
~This is draft while I test (I suspect CI may fail initially because I had problems with credentials locally).~
- The Esri credentials we had for omgeo testing seemed to have disappeared, so I created some new ones and updated the ones we have
- This is heavily cribbed from [the setup for django-ecsmanage](https://github.com/azavea/django-ecsmanage/tree/develop/.github/workflows).

## Testing instructions

- Push a dummy commit to this branch and confirm that it triggers a build (and that the build succeeds).
- I'm not sure whether we can fully test the release flow until this is merged to `develop`, but I believe the path to test should be as follows:
  - Register on [TestPyPi](https://test.pypi.org/account/register/) and get account credentials (this is probably best done by someone with easy access to the email we use for company-wide accounts).
  - Put the account credentials into the appropriate secrets for this repository
  - Push a dummy tag and confirm that a release is made to TestPyPi
  - Confirm you can [install the new release from TestPyPi](https://packaging.python.org/tutorials/packaging-projects/#installing-your-newly-uploaded-package)

Assuming the above is completed, the following cleanup tasks also need to be done before merging:
- [x] Update the repository secrets to point to production PyPi
- [x] Remove the `Delete me!` commit from this branch / develop
- [x] Delete the test tag
